src/net.c

```diff
@@ -777,6 +861,16 @@ static int hyper_setup_interface(struct rtnl_handle *rth,
 	req.n.nlmsg_type = RTM_NEWADDR;
 	req.ifa.ifa_family = AF_INET;
 
+	// This is where we get the device name from MAC address in case
+	// device name is not provided.
+	if (!iface->device &&
+	    hyper_get_iface_name_from_mac_addr(iface->mac_addr,
+					       &iface->device)) {
+		fprintf(stderr, "failed to get interface name from MAC"
+			" address %s\n", iface->mac_addr);
+		return -1;
+	}
+
```

Comment:
@sboeuf can we add another check here to check is mac_address matches the one obtained using device name in case both device name and mac address are provided. It will offer a failsafe just in case something goes wrong on the host side.